xlml/apis/task.py

```diff
@@ -38,6 +41,72 @@ def run() -> DAGNode:
     ...
 
 
+@dataclasses.dataclass
+class DockerBuildTask(BaseTask):
+  """This is a class to set up a Docker build task for XPK workloads.
+
+  XPK runs on Kubernetes, which requires a docker image to execute the workload.
+  To ensure freshness of the images and reduce unnecessary setup on workers, a
+  single BashOperator runs the docker build to prepare the image for execution,
+  then all workers will use the resulting image for their containers.
+
+  The resulting image can be shared across multiple tasks within the DAG, and
+  the unique image name is accessible as the output of the DockerBuildTask.
+
+  Attributes:
+    build_dir: Relative path within the repository to run the build. This path
+        must contain a Dockerfile or cloudbuild.yaml.
+    image_name: The base image name to build. The timestamp of the build will
+        be used to tag the image.
+    custom_build: When set, run the build using cloudbuild.yaml. The
+        ``_IMAGE_NAME`` substitution will be the tagged docker image which will
+        be the output of this AirFlow task. Defaults to False, meaning the build
+        is a direct docker build.
+    task_id: Task ID to use for the operator. Defaults to
+        ``DockerBuildTask.TASK_ID``.
+  """
+
+  build_dir: str
+  image_name: str
+  custom_build: bool = False
+  task_id: Optional[str] = None
+
+  # Path on the Airflow worker which contains the Airflow GCS bucket contents.
+  GCS_PATH = "/home/airflow/gcs/dags"
+  DEFAULT_TASK_ID = "docker_image_build"
+  IMAGE_NAME_SUBSTITUTION = "_IMAGE_NAME"
+  BUILD_SCRIPT = """
+  set -xe
+  cd {build_path}
+  gcloud builds submit --machine-type=e2-highcpu-32 {build_flags} .
```

Comment:
For machine-type, are we able to define the option like [this](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/pipeline/auto-push.cloudbuild.yml#L27)?

Reply:
We could, but that requires a `cloudbuild.yaml`. This way, the regular docker builds will also use a beefier host for the builds.